src/classes/Pantry.js

```diff
@@ -1,46 +1,72 @@
-  //   let userIngredientsAmt = this.shelf.reduce((allUserIngredientAmt, userIngredient) => {
-  //     recipe.ingredients.forEach(recipeIngredient => {
-  //       if (recipeIngredient.id === userIngredient.ingredient)
-  //         allUserIngredientAmt.push(userIngredient.amount)
-  //     })
-  //     return allUserIngredientAmt;
-  //   }, [])
-  //   return userIngredientsAmt
-  // }
-
-  determineAmountOfIngredientsUserHas() {
-    let userIngredientAmt = this.shelf.map(ingredient => {
-      return ingredient.amount 
+  showPantryIngredientInfo() {
+    this.shelf.map(userIngredient => {
+      ingredientsData.forEach(ingredient => {
+        if (ingredient.id === userIngredient.ingredient) {
+          userIngredient.name = ingredient.name
+        }
+      })
     })
-    return userIngredientAmt;
+    return this.shelf
   }
-  //user ingredients amount is an array of the amount of each ingredient the user
-  //has if it matches the ingredients in the recipe
-
+  //Could we use the findRecipeIngredientInfo method from the Recipe class to instead of repeating the logic here?
   determineIfEnoughIngredientsForRecipe(recipe) {
-    let userIngredientAmt = this.determineAmountOfIngredientsUserHas(recipe);
-    let userMessage = "";
-    console.log(userIngredientAmt);
-    userIngredientAmt.forEach(amount => {
-      recipe.ingredients.forEach(ingredient => {
-        if(amount < ingredient.quantity.amount || !userIngredientAmt.includes(ingredient.quantity.amount)) {
-          userMessage = "You don't have enough ingredients to cook this recipe"
-        } else {
-          userMessage = "Let's get cookin', good lookin";
+    let pantryIngredients = this.showPantryIngredientInfo();
+    let recipeIngredients = recipe.findRecipeIngredientInfo();
+    const ingredientsNeeded = recipeIngredients.filter(ingredient => {
+      const enoughIngredients = pantryIngredients.every(item => ingredient.id === item.id && item.amount >= parseInt(ingredient.quantity.amount));
+      if (!enoughIngredients) {
+        return ingredient
+        //returns an array of recipe ingredient objects that are missing from the pantry or less than the amount needed to make the recipe
+      }
+    })
+    return ingredientsNeeded
+  }
+
+  findAmountNeeded(recipe) {
+    let recipeIngredients = recipe.findRecipeIngredientInfo();
+    const newShelf = recipeIngredients.map(recipeIngredient => {
+      this.shelf.forEach(pantryIngredient => {
+        if (recipeIngredient.id === pantryIngredient.ingredient) {
+          recipeIngredient.amountNeeded = pantryIngredient.amount + (parseInt(recipeIngredient.quantity.amount) - pantryIngredient.amount)
+        }
+        if (recipeIngredient.id !== pantryIngredient.ingredient) {
+          recipeIngredient.amountNeeded = recipeIngredient.quantity.amount
         }
       })
+        return recipeIngredient
     })
-    return userMessage
+    return newShelf.filter(ingredient => ingredient.amountNeeded)
```

Comment:
This was cool to figure out how to do, and figure out how to get the information we wanted out of it.